examples/cpp/dds/AdvanceConfigurationExample/AdvanceConfiguration_main.cpp

```diff
@@ -0,0 +1,272 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file AdvanceConfiguration_main.cpp
+ *
+ */
+
+#include <string>
+
+#include "arg_configuration.h"
+#include "AdvanceConfigurationPublisher.h"
+#include "AdvanceConfigurationSubscriber.h"
+#include "types.hpp"
+
+enum EntityType
+{
+    PUBLISHER,
+    SUBSCRIBER
+};
+
+int main(
+        int argc,
+        char** argv)
+{
+    int columns;
+
+#if defined(_WIN32)
+    char* buf = nullptr;
+    size_t sz = 0;
+    if (_dupenv_s(&buf, &sz, "COLUMNS") == 0 && buf != nullptr)
+    {
+        columns = strtol(buf, nullptr, 10);
+        free(buf);
+    }
+    else
+    {
+        columns = 80;
+    }
+#else
+    columns = getenv("COLUMNS") ? atoi(getenv("COLUMNS")) : 80;
+#endif // if defined(_WIN32)
+
+    EntityType type = PUBLISHER;
+    std::string topic_name = "HelloWorldTopic";
+    int count = 0;
+    long sleep = 100;
+    int num_wait_matched = 0;
+    int domain = 0;
+    bool async = false;
+    TransportType transport = DEFAULT;
+    int hops = -1;
+    bool reliable = false;
+    bool transient = false; // transient local
+    std::string partitions = "";
+    bool use_ownership = false;
+    unsigned int ownership_strength = 0;
```

Comment:
Please check whether the passed values are negative. In fact, this should also be done for all numerical arguments (domain, interval, etc.)

Reply:
Already done in https://github.com/eProsima/Fast-DDS/pull/2951